docs/_layouts/global.html

```diff
@@ -207,14 +205,84 @@ <h1 class="title">{{ page.title }}</h1>
     })
 </script>
 
+<!-- Pure JavaScript implementation of ScrollSpy -->
 <script>
-    var controller = new ScrollMagic.Controller();
+    // finds toc and automatically sorts by main list items as parents and nested list items as children
 
-    new ScrollMagic.Scene({
-        offset: 20
-    })
-    .setClassToggle(".navbar-inner", "shrunk")
-    .addTo(controller);
+    var toc = document.getElementById("toc");
+    if (toc != null) { // checks to see if table of contents exists in page
+        var topPadding = 12; // amount of space above header to start noticing a section
+        var navBar = document.getElementById("topbar").offsetHeight;
+        var tocEl = toc.nextElementSibling.getElementsByTagName("li"); // table of contents list
+        var sections = toc.nextElementSibling.getElementsByTagName("a"); // table of contents links
+        sections = Array.prototype.slice.call(sections);
+        var sectionsLen = sections.length;
+        var parentIndex;
+        var parentTag = sections[0].getAttribute("href") == "#" ? "H1" : document.getElementById(sections[0].getAttribute("href").substring(1)).nodeName;
+        for (var i = 0; i < sectionsLen; i++) {
+            // goes through links and creates a 2d array with:
+            // [element, offset from top, group]
+            var element = sections[i].getAttribute("href") == "#" ? document.getElementById("content").firstElementChild : document.getElementById(sections[i].getAttribute("href").substring(1));
+            if (element.nodeName == parentTag) {
+                parentIndex = i;
+            }
+            sections[i] = [element, element.offsetTop - navBar - topPadding, parentIndex];
+        }
+        var active = 0;
+        tocEl[0].classList.toggle("active");
+        var sublist;
+        var sublistPos = 0;
+        var sublistChildren = 0;
+        var locked = false;
+        var runOnScroll =  function(e) {
+            if (locked) return; // if still scrolling, do not run
+            locked = true;
+            var pos = 0;
+
+            // get current active element
+            for (var i = 0; i < sectionsLen; i++) {
+                if (window.scrollY > sections[i][1]) {
+                    pos = i;
+                }
+                else {
```

Comment:
Can you put this line on the previous line? Thanks.

if () {
...
} else { 
...
}